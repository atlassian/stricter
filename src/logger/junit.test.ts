import logJunit from './junit';

describe('junitLogger', () => {
    const logMock = jest.fn();
    console.log = logMock;

    beforeEach(() => {
        logMock.mockReset();
    });

    it('outputs valid empty xml', () => {
        logJunit([]);
        const output = logMock.mock.calls[0][0];
        expect(output).toMatchSnapshot();
    });

    it('outputs valid complex xml', () => {
        logJunit([
            {
                rule: 'rule1',
                errors: [
                    'error1',
                    `someting 'that' "needs" <escaping> and closes it's CDATA ]]> early`,
                ],
            },
            { rule: 'rule2', warnings: ['rule2-warning'] },
            { rule: 'rule3', errors: ['rule3-error'], warnings: ['rule3-warning'] },
        ]);
        const output = logMock.mock.calls[0][0];
        expect(output).toMatchSnapshot();
    });
});
