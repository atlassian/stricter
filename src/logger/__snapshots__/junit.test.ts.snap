// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`junitLogger outputs valid complex xml 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
    <testsuites package=\\"stricter\\">
        <testsuite name=\\"stricter\\" failures=\\"5\\" errors=\\"3\\" tests=\\"3\\">
    <testcase name=\\"rule1\\">
    <failure type=\\"error\\"><![CDATA[
error1
someting 'that' \\"needs\\" <escaping> and closes it's CDATA ]]]]><![CDATA[> early]]></failure>
</testcase>
<testcase name=\\"rule2\\">
    <failure type=\\"warning\\"><![CDATA[
rule2-warning]]></failure>
</testcase>
<testcase name=\\"rule3\\">
    <failure type=\\"error\\"><![CDATA[
rule3-error]]></failure>
<failure type=\\"warning\\"><![CDATA[
rule3-warning]]></failure>
</testcase>
</testsuite>
    </testsuites>"
`;

exports[`junitLogger outputs valid empty xml 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
    <testsuites package=\\"stricter\\">
        <testsuite name=\\"stricter\\" failures=\\"0\\" errors=\\"0\\" tests=\\"1\\">
    <testcase name=\\"no errors\\">
    
</testcase>
</testsuite>
    </testsuites>"
`;
